lib/semian/circuit_breaker.rb

```diff
@@ -87,7 +87,7 @@ def open
 
     def half_open
       log_state_transition(:half_open)
-      @state.half_open
+      half_open!
```

Comment:
IMO this delegator should just be removed, and `State#half_open!` can then be renamed back to `State#half_open`.